src/ParamHandler.ts

```diff
@@ -47,10 +47,10 @@ export class ParamHandler {
         if (param.isRequired) {
             // todo: make better error messages here
             if (param.name && isValueEmpty) {
-                return Promise.reject(new ParameterRequiredError(request.url, request.method, param.name));
+                throw new ParameterRequiredError(request.url, request.method, param.name);
```

Comment:
You have misunderstood me... I just explained why you have to use `Promise.reject` because throwed errors will not be handled properly. See [here](https://github.com/iH8c0ff33/routing-controllers/blob/master/src/RoutingControllerExecutor.ts#L81) that there is `.catch()` chaining on promise, to sync try-catch block.